loleaflet/Makefile.am

```diff
@@ -519,13 +519,12 @@ node_modules: package.json archived-packages
 	@npm install
 	@touch node_modules
 
-Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status node_modules
-	@case '$?' in \
-	 *config.status*) \
-	  cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh;; \
-	 *) \
-	  cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@;; \
-	esac;
+define target_makefile
+Makefile: $$(srcdir)/Makefile.in $$(top_builddir)/config.status $(1)
```

Comment:
I have never seen Makefile target defined in Makefile.am. This is managed by automake and autoconf. 

At least some comment explaining what's going on is necessary here. Why are we defining Makefile rule? why can't we have node_modules as a dependency for the rules that depend on it? Why is node_modules an argument? And why do we explicitly invoke *top-level* make ?! (That looks so wrong--it's recursive!)